Fix(edit some UI and some functionality)

mohsinayaz opened the pull request:
1:- Edit some UI changes according to the instructions and add some new things also.
2:- Remove the procedure to pass data form URL. Get user by its id and show the user details on page

Commit:
Fix(edit some UI and some functionality)

1:- Edit some UI changes according to the instructions and add some new things also.
2:- Remove the procedure to pass data form URL. Get user by its id and show the user details on page

AhsanAyaz commented on src/app/home/home.component.scss:
```diff
@@ -1,29 +1,56 @@
-            width: 136px;
-            border-radius: 50%;
+            width: 500px;
+            border: 1px solid #00000f;
```
Would have been nice to use constants for scss variables.

mohsinayaz replied:
done

AhsanAyaz commented on src/app/home/home.component.scss:
```diff
@@ -1,29 +1,56 @@
+                width: 100%;
+                height: 74px;
+                top: 0;
+                background: #f9f9f9;
```
Same here. Could have used a constant.

mohsinayaz replied:
done

mohsinayaz replied:
done

AhsanAyaz commented on src/app/user-detail/user-detail.component.ts:
```diff
@@ -1,32 +1,39 @@
-      this.user = JSON.parse(params.data);
+    let userId: number;
+    this.route.params.subscribe((params:Params) => {
+      userId = +params['userId'];
```
A good practice is to use `userId = Number(params['userId']);` instead

mohsinayaz replied:
done

AhsanAyaz commented on src/app/user-detail/user-detail.component.ts:
```diff
@@ -1,32 +1,39 @@
-    this.detail = this.user.name;
-    this.title = 'Hi, My name is';
+    if (!this.userService.newData) {
+      this.userService.getUsers().subscribe(data => {
```
We're not using the `data` variable at all. We can just use `() =>` instead.

mohsinayaz replied:
done

AhsanAyaz reviewed (changes requested):
Looks good overall 🎉 Great job.
Just a few comments

Commit:
fix(minor-changes)

1:- Fix some UI changes
2:- apply the changes that are commented on previous commit
